lib/internal/streams/end-of-stream.js

```diff
@@ -26,6 +26,11 @@ function eos(stream, opts, callback) {
     throw new ERR_INVALID_ARG_TYPE('callback', 'function', callback);
   }
 
+  if (stream.destroyed || stream.closed) {
```

Comment:
Is `closed` part of the published stream contract? In the test, are we checking both these flags - I don't think so.

Reply:
> Is closed part of the published stream contract?

Nope, though some streamlike objects use it (or might use it in the future). https://github.com/nodejs/node/pull/28621

It's a nice to have I guess...

Reply:
would you mind to drop it then, and maybe we should assess to add it if it's needed later on? It might have unwanted consequences.